torch/distributions/transforms.py

```diff
@@ -652,9 +657,9 @@ def log_abs_det_jacobian(self, x, y):
         for trans, length in zip(self.transforms, self.lengths):
             xslice = x.narrow(self.dim, start, length)
             yslice = y.narrow(self.dim, start, length)
-            logdetjacs.append(trans.log_abs_det_jacobian(xslice, yslice))
+            logdetjacs.append(trans.log_abs_det_jacobian(xslice, yslice).view(x.shape + (1,) * self.event_dim))
             start = start + length  # avoid += for jit compat
-        return torch.cat(logdetjacs, dim=self.dim)
+        return torch.cat(logdetjacs, dim=self.dim)._sum_rightmost(self.event_dim)
```

Comment:
Is `._sum_rightmost()` a method? I would think you'd need `_sum_rightmost(torch.cat(...), self.event_dim)`.

Reply:
Yes, sorry let me make that change